gabarit/template_nlp/nlp_project/package_name/models_training/models_tensorflow/model_keras.py

```diff
@@ -353,27 +355,44 @@ def predict_proba(self, x_test, alternative_version: bool = False, **kwargs) ->
         x_test = self._prepare_x_test(x_test)
         # Process
         if alternative_version:
-            return self._alternative_predict_proba(x_test)
+            return self._alternative_predict_proba(x_test, batch_size=batch_size)
         else:
             # We advise you to avoid using `model.predict` with newest TensorFlow versions (possible memory leak) in a production environment (e.g. API)
             # https://github.com/tensorflow/tensorflow/issues/58676
             # Instead, you can use the alternative version that uses tf.function decorator & model.__call__
             # However, it should still be better to use `model.predict` for one-shot, batch mode, large input, iterations.
-            return self.model.predict(x_test, batch_size=128, verbose=1)  # type: ignore
+            return self.model.predict(x_test, batch_size=batch_size, verbose=1)  # type: ignore
 
     @utils.trained_needed
-    def _alternative_predict_proba(self, x_test, **kwargs) -> np.ndarray:
+    def _alternative_predict_proba(self, x_test, batch_size: int = 128, **kwargs) -> np.ndarray:
         '''Predicts probabilities on the test dataset - Alternative version
         Should be faster with low nb of inputs.
 
         Args:
             x_test (?): Array-like or sparse matrix, shape = [n_samples]
+        Kwargs:
+            batch_size (int): size (approximate) of batches
         Returns:
             (np.ndarray): Array, shape = [n_samples, n_classes]
         '''
-        return self._serve(x_test).numpy()
+        # Assert batch size is >= 1
+        batch_size = max(1, batch_size)
+        # Process by batches - avoid huge memory impact
+        nb_batches = max(1, len(x_test)//batch_size)
+        np_results = None
+        for arr in np.array_split(x_test, nb_batches, axis=0):
```

Comment:
It is faster (but use more memory) to use this loop to generate a list of array that we concatenate at the end instead of using concatenate at each step (and avoid using a "if np_results is None" which makes the code more concise